extensions/contract-termination/src/test/java/de/sovity/edc/extension/contacttermination/query/TerminateContractQueryTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright (c) 2024 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.extension.contacttermination.query;
+
+import de.sovity.edc.extension.contacttermination.ContractTermination;
+import de.sovity.edc.extension.e2e.db.EdcRuntimeExtensionWithTestDatabase;
+import lombok.val;
+import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiation;
+import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiationStates;
+import org.jooq.DSLContext;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import java.time.OffsetDateTime;
+
+import static de.sovity.edc.ext.db.jooq.enums.ContractTerminatedBy.COUNTERPARTY;
+import static de.sovity.edc.extension.e2e.connector.config.ConnectorConfigFactory.forTestDatabase;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class TerminateContractQueryTest {
+
+    @RegisterExtension
+    static EdcRuntimeExtensionWithTestDatabase providerExtension = new EdcRuntimeExtensionWithTestDatabase(
+        ":launchers:connectors:sovity-dev",
```

Comment:
don't these tests always have to be in separate modules so that they can depend on said module to be built?

Reply:
also in CI, right?

Reply:
Yes, the tests are green on GH too.
And this is close to the config that the connector will run with in prod.
I'll just keep it like this until I see a problem.